cmd/subctl/recover_broker_info.go

```diff
@@ -0,0 +1,171 @@
+			return recoverRestConfigProducer.RunOnSelectedContext(
+				func(submCluster *cluster.Info, namespace string, status reporter.Interface) error {
+					if isSubmJoinedToBroker(clusters, submCluster) {
+						status.Success("Found a Submariner installation on cluster %q joined to the Broker", submCluster.Name)
+						return broker.RecoverData(brokerCluster, submCluster, brokerObj, namespace, status)
+					}
+
+					return status.Error(
+						fmt.Errorf("submariner is not installed on cluster %s. "+
+							"Please specify the cluster where Submariner is installed via `--kubeconfig` or `--context` flag"+
+							"", submCluster.Name), "")
+				}, status)
+		}, status)
+
+	if contextFound || err != nil {
+		return err //nolint:wrapcheck // No need to wrap errors here.
+	}
+
+	// if --brokerconfig not provided, search for broker on current context and proceed to get Submariner
+	return recoverRestConfigProducer.RunOnSelectedContext(
+		func(clusterInfo *cluster.Info, namespace string, status reporter.Interface) error {
+			brokerObj, err := getBroker(clusterInfo, namespace, "Please specify the cluster where the Broker is installed "+
+				"via the "+"`--brokerconfig` or `--brokercontext` flag and the namespace via the --brokernamespace flag", status)
+			if err != nil {
+				return err
+			}
+
+			clusters, err := clusterInfo.GetClusters(namespace)
+			if err != nil {
+				return status.Error(err, "error listing joined clusters")
+			}
+
+			ok, err := tryToRecoverFromBroker(clusterInfo, brokerObj, namespace, clusters, status)
+			if ok || err != nil {
+				return err
+			}
+
+			return status.Error(
+				fmt.Errorf("submariner is not installed on cluster %q. "+
+					"Please specify the cluster where Submariner is installed via `--kubeconfig` or `--context` flag"+
+					"", clusterInfo.Name), "")
+		}, status)
+}
+
+func getBroker(clusterInfo *cluster.Info, namespace, notFoundMsg string, status reporter.Interface) (*v1alpha1.Broker, error) {
+	status.Start("Checking if the Broker is installed on cluster %q in namespace %q", clusterInfo.Name, namespace)
+
+	brokerObj, err := clusterInfo.GetBroker(namespace)
+	if apierrors.IsNotFound(err) {
+		return nil, status.Error(fmt.Errorf("the Broker is not installed on the specified cluster. %s", notFoundMsg), "")
+	}
+
+	if err != nil {
+		return nil, status.Error(err, "")
+	}
+
+	status.End()
+
+	return brokerObj, nil
+}
+
+func tryToRecoverFromBroker(brokerCluster *cluster.Info, brokerObj *v1alpha1.Broker,
+	brokerNamespace string, clusters []submarinerv1.Cluster, status reporter.Interface,
+) (bool, error) {
+	status.Start("Checking if there are any cluster(s) joined to the Broker")
+
+	if len(clusters) == 0 {
+		return false, status.Error(
+			errors.New(
+				"no clusters are joined to the Broker. Please re-run the `deploy-broker` command to regenerate the broker-info."+
+					"subm file"), "")
+	}
+
+	status.Success("Found cluster(s) joined to the Broker")
```

Comment:
Similar to a comment elsewhere, I think the green checkmark is a sufficient positive reply unless we provide more info in the message.  So I think either remove this message or change it to:

```suggestion
	status.Success("Found %d cluster(s) joined to the Broker", len(clusters))
```